smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/SwiftSymbolProvider.kt

```diff
@@ -284,6 +296,110 @@ class SwiftSymbolProvider(private val model: Model, val swiftSettings: SwiftSett
+            }
+            is IntEnumShape -> {
+                // Get the corresponding enum member name (enum case name) for the int value from default trait
+                val enumMemberName = targetShape.enumValues.entries.firstOrNull {
+                    it.value == defaultValueLiteral.toInt()
+                }!!.key
+                resolvedDefaultValue = ".${swiftEnumCaseName(enumMemberName, defaultValueLiteral)}"
+            }
+            is StringShape -> {
+                resolvedDefaultValue = "\"$defaultValueLiteral\"" // Swift string literal
+            }
+            is MapShape -> {
+                resolvedDefaultValue = "[:]" // Empty dictionary is the only valid default value
+            }
+            is BlobShape -> {
+                builder.putProperty("NeedsDataImport", FoundationTypes.Data)
+                resolvedDefaultValue = if (targetShape.hasTrait<StreamingTrait>()) {
+                    "ByteStream.data(Data(\"$defaultValueLiteral\".utf8))"
+                } else {
+                    "Data(\"$defaultValueLiteral\".utf8)"
+                }
+            }
+            is DocumentShape -> {
+                val node = shape.getTrait<DefaultTrait>()!!.toNode()
+                when {
+                    node.isObjectNode -> {
+                        resolvedDefaultValue = "Document.object([:])" // Empty map is the only valid default value
+                    }
+                    node.isArrayNode -> {
+                        resolvedDefaultValue = "Document.array([])" // Empty array is the only valid default value
+                    }
+                    node.isBooleanNode -> {
+                        resolvedDefaultValue = "Document.boolean($defaultValueLiteral)"
+                    }
+                    node.isStringNode -> {
+                        resolvedDefaultValue = "Document.string(\"$defaultValueLiteral\")"
+                    }
+                    node.isNumberNode -> {
+                        resolvedDefaultValue = "Document.number($defaultValueLiteral)"
+                    }
+                }
+            }
+            is TimestampShape -> {
+                resolvedDefaultValue = "Date(timeIntervalSince1970: $defaultValueLiteral)"
+            }
+            is FloatShape, is DoubleShape -> {
+                val decimal = ".0".takeIf { !defaultValueLiteral.contains(".") } ?: ""
+                resolvedDefaultValue = defaultValueLiteral + decimal
+            }
+            else -> {
+                /*
+                 For: boolean, byte, short, integer, long, bigInteger, bigDecimal
+                    just take the literal string value from the trait.
+                 */
+                resolvedDefaultValue = defaultValueLiteral
+            }
+        }
+
+        // Return Symbol.Builder with resolved default value saved to Symbol.Builder's property bag
+        return builder.defaultValue(resolvedDefaultValue)
+    }
```

Comment:
Should refactor this to make use of `when` as an expression in kotlin

example:
```
val resolvedDefaultValue = when (targetShape) {
    is ListShape -> "[]"
    is EnumShape -> ".${swiftEnumCaseName(null, defaultValueLiteral)}"
    ... (etc)
}
```

Same applies to the when inside of DocumentationShape. Since this is also a nested when consider moving the logic to handle document shape to a helper function

Reply:
Refactored as suggested 👍